pyproject.toml

```
@@ -68,11 +68,12 @@ test = [
     "coverage",
 ]
 docs = [
-    "sphinx>=1.6,<8",
+    "sphinx>6",
     "docutils<0.22",
     "sphinx-automodapi",
     "sphinx-design",
-    "sphinx_rtd_theme==2.0.0",
+    "sphinx_rtd_theme==3.0.0",
```

Any reason to peg at `3.0.0` and not allow it to float `>=3`?

---

Yeah, the RTD theme can be picky about the docutils and sphinx versions.  So rather than being caught out by those, I have typically pinned this package to a specific version and keep track of new releases.  It's more maintenance overhead, but the docs can fail pretty silently (i.e., someone has to notice that the webpages aren't right) and this is my attempt to limit those silent failures.

There's a way we can add a doc compilation check to the GitHub CI (e.g., astropy does it), but I haven't figured out how to implement that yet.